src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -305,32 +346,15 @@ class OrganizationIntegrations extends AsyncComponent<
 
   renderBody() {
     const {orgId} = this.props.params;
-    const {reloading, orgOwnedApps, publishedApps, extraApp, plugins} = this.state;
-    const published = publishedApps || [];
-    // If we have an extra app in state from query parameter, add it as org owned app
-    if (extraApp) {
-      orgOwnedApps.push(extraApp);
-    }
+    const {reloading, list, displayedList} = this.state;
 
-    // we dont want the app to render twice if its the org that created
-    // the published app.
-    const orgOwned = orgOwnedApps.filter(app => {
-      return !published.find(p => p.slug === app.slug);
+    const fuse = new Fuse(list, {
```

Comment:
@NisanthanNanthakumar We have a function called `createFuzzySearch` which is a wrapper around `new Fuse`. Let's use that instead.

Reply:
Also, is there a reason why we are calling this in the `renderBody` method? Could we just call this in `onChange` callback?

Reply:
@NisanthanNanthakumar What about using `createFuzzySearch`?

Reply:
@scefali missed this